src/sql/azdata.proposed.d.ts

```diff
@@ -517,6 +537,13 @@ declare module 'azdata' {
 		 * @returns The new password that is returned from the operation or undefined if unsuccessful.
 		 */
 		export function openChangePasswordDialog(profile: IConnectionProfile): Thenable<string | undefined>;
+
+		/**
+		 * Opens the error dialog with customization options provided.
+		 * @param options Dialog options to customize error dialog.
+		 * @returns Id of action button clicked by user, e.g. ok, cancel
+		 */
+		export function openCustomErrorDialog(options: window.ICustomDialogOptions): Thenable<string | undefined>;
```

Comment:
I'd put this in the window namespace instead of connection. This would be useful for other scenarios as well.

Reply:
Done